library/std/src/sys/pal/unix/thread.rs

```diff
@@ -296,6 +296,63 @@ impl Thread {
         }
     }
 
+    // Any unix that has clock_nanosleep
+    #[cfg(any(
+        target_os = "freebsd",
+        target_os = "netbsd",
+        target_os = "linux",
+        target_os = "android",
+        target_os = "solaris",
+        target_os = "illumos",
+        target_os = "dragonfly",
+        target_os = "hurd",
+        target_os = "fuchsia",
+        target_os = "vxworks",
+    ))]
+    pub fn sleep_until(deadline: Instant) {
+        let mut ts = deadline
+            .into_inner()
+            .into_timespec()
+            .to_timespec()
+            .expect("Timespec is narrower then libc::timespec thus conversion can't fail");
```

Comment:
What do you mean by this? Couldn't targets with 32-bit `time_t` fail? That's why `sleep` has to loop, and I guess you can just fallback to that when the deadline `timespec` isn't representable.

Reply:
good point, I might have forgotten about 32 bit when I wrote this :sweat_smile: